coderd/prometheusmetrics/prometheusmetrics.go

```diff
@@ -110,16 +118,53 @@ func Workspaces(ctx context.Context, registerer prometheus.Registerer, db databa
 		}
 		jobs, err := db.GetProvisionerJobsByIDs(ctx, jobIDs)
 		if err != nil {
+			ids := make([]string, 0, len(jobIDs))
+			for _, id := range jobIDs {
+				ids = append(ids, id.String())
+			}
+
+			logger.Warn(ctx, "failed to load provisioner jobs", slog.F("ids", ids), slog.Error(err))
 			return
 		}
 
-		gauge.Reset()
+		workspaceStatusTotals.Reset()
 		for _, job := range jobs {
 			status := codersdk.ProvisionerJobStatus(job.JobStatus)
-			gauge.WithLabelValues(string(status)).Add(1)
+			workspaceStatusTotals.WithLabelValues(string(status)).Add(1)
 		}
 	}
 
+	updateWorkspaceDetails := func() {
+		ws, err := db.GetWorkspaces(ctx, database.GetWorkspacesParams{
+			Deleted:     false,
+			WithSummary: false,
+		})
+		if err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
+				// clear all series if there are no database entries
+				workspaceStatuses.Reset()
+			}
+
+			logger.Warn(ctx, "failed to load active workspaces", slog.Error(err))
```

Comment:
nit: maybe `logger.Error`?

Reply:
I wouldn't strictly call it an error because it could be an instance of `sql.ErrNoRows` which is valid but undesirable. To be the most correct I would make it warn in case of `sql.ErrNoRows` and error in all other cases, but that feels unnecessary.

Reply:
Believe so, yes.